File: app/src/main/java/im/tox/antox/LeftPaneItem.java
package im.tox.antox;

/**
 * Created by ollie on 04/03/14.
 */
public class LeftPaneItem {

    public int viewType; //Either a header, request, or contact
    public String first; //Either the header, the request key, or the contact name
    public String second; //Either null, the request message, or the contact status message
    public int icon; //Null unless friend, if friend then icon status

    public LeftPaneItem() {
        super();
    }

    public LeftPaneItem(int viewType, String first, String second, int icon) {
        super();
        this.viewType = viewType;
        this.first = first;
        this.second = second;
        this.icon = icon;
    }

    public String first() {
        return first;
    }
    public String second() {
        return second;
    }
    public int icon() {
        return icon;
    }
    public int viewType() {
        return viewType;
    }
}

File: app/src/main/scala/chat/tox/antox/utils/LeftPaneItem.scala
package chat.tox.antox.utils

import java.io.File
import java.sql.Timestamp

import chat.tox.antox.fragments.ContactItemType.ContactItemType
import chat.tox.antox.wrapper.ContactKey
import im.tox.tox4j.core.enums.ToxUserStatus

class LeftPaneItem(
                    val viewType: ContactItemType,
                    val key: ContactKey,
                    val image: Option[File],
                    val first: String, // name
                    val second: String, // status message, last message or call info depending on which tab
                    val secondImage: Option[Int],
                    val isOnline: Boolean,
                    val status: ToxUserStatus,
                    val favorite: Boolean,
                    val count: Int,
                    val timestamp: Timestamp, // if call is active this is start time
                    val activeCall: Boolean) {

  def this(viewType: ContactItemType, key: ContactKey, message: String) =
    this(viewType, key, None, key.toString, message, None, false, null, false, 0, null, false)

}